src/Toolkit/Toolkit/UI/Controls/OverviewMap/OverviewMapController.cs

```diff
@@ -0,0 +1,212 @@
+
+            if (_connectedView is MapView mv && mv.VisibleArea is Polygon area)
+            {
+                _extentOverlay.Graphics.Add(new Graphic(area));
+            }
+            else if (_connectedView is SceneView sv && sv.GetCurrentViewpoint(ViewpointType.CenterAndScale)?.TargetGeometry is MapPoint centerPoint)
+            {
+                _extentOverlay.Graphics.Add(new Graphic(centerPoint));
+            }
+        }
+
+        private void OnViewpointChanged(object? sender, EventArgs e)
+        {
+            GeoView? sendingView = sender as GeoView;
+            GeoView? receivingView = sendingView == _overview ? _connectedView : _overview;
+
+            // Check for IsNavigating is intended to ignore OnViewpointChanged calls triggered by the call to SetViewpoint.
+            if ((sendingView?.IsNavigating ?? false) && receivingView != null)
+            {
+                ApplyViewpoint(sendingView, receivingView);
+                return;
+            }
+
+            if (_connectedView != null && !_hasSetViewpoint)
+            {
+                ApplyViewpoint(_connectedView, _overview);
+            }
+        }
+
+        private void OnNavigationCompleted(object? sender, EventArgs e)
+        {
+            GeoView? sendingView = sender as GeoView;
+            GeoView? receivingView = sendingView == _overview ? _connectedView : _overview;
+            if (receivingView != null && sendingView != null)
+            {
+                ApplyViewpoint(sendingView, receivingView);
+            }
+        }
+
+        private void ApplyViewpoint(GeoView sendingView, GeoView receivingView)
+        {
+            if (sendingView.GetCurrentViewpoint(ViewpointType.CenterAndScale) is Viewpoint existingViewpoint)
+            {
+                _hasSetViewpoint = true;
+                var scaleMultiplier = sendingView == _overview ? 1.0 / ScaleFactor : ScaleFactor;
+                Viewpoint newViewpoint = new Viewpoint((MapPoint)existingViewpoint.TargetGeometry, existingViewpoint.TargetScale * scaleMultiplier);
+                receivingView.SetViewpoint(newViewpoint);
+                UpdateGraphic();
+            }
+        }
+
+        private void UpdateSymbol()
+        {
+            if (Symbol != null)
+            {
+                _extentOverlay.Renderer = new SimpleRenderer(Symbol);
+            }
+            else if (_connectedView is MapView)
+            {
+                _extentOverlay.Renderer = new SimpleRenderer(
+                    new SimpleFillSymbol(SimpleFillSymbolStyle.Null, System.Drawing.Color.Transparent,
+                    new SimpleLineSymbol(SimpleLineSymbolStyle.Solid, System.Drawing.Color.Red, 1)));
+            }
+            else if (_connectedView is SceneView)
+            {
+                _extentOverlay.Renderer = new SimpleRenderer(
+                    new SimpleMarkerSymbol(SimpleMarkerSymbolStyle.Cross, System.Drawing.Color.Red, 16));
```

Comment:
Consider making the default symbols a private static variable for reuse (unless you start exposing both as properties, when this wouldn't be needed)